pkg/ccl/testccl/sqlstatsccl/sqlstatsccl_test: TestSQLStatsRegions failed

pkg/ccl/testccl/sqlstatsccl/sqlstatsccl_test.TestSQLStatsRegions [failed](https://mesolite.cluster.engflow.com/invocations/default/d2ef973d-64c6-40bb-b4c4-cefc2fdf1d4b?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3Fsc3RhdHNjY2w6c3Fsc3RhdHNjY2xfdGVzdA==) on master @ [3878bf6a9173e1654985c8fafc198b82fc552eac](https://github.com/cockroachdb/cockroach/commits/3878bf6a9173e1654985c8fafc198b82fc552eac):


```
=== RUN   TestSQLStatsRegions
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestSQLStatsRegions1089757201
    test_log_scope.go:76: use -show-logs to present logs inline
    sql_stats_test.go:211: -- test log scope end --
test logs left over in: outputs.zip/logTestSQLStatsRegions1089757201
--- FAIL: TestSQLStatsRegions (90.78s)
=== RUN   TestSQLStatsRegions/secondary_tenant
    --- FAIL: TestSQLStatsRegions/secondary_tenant (9.96s)
=== RUN   TestSQLStatsRegions/secondary_tenant/distsql=false
    sql_stats_test.go:205: 
        	Error Trace:	pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:205
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:148
        	            				pkg/testutils/subtest.go:23
        	Error:      	Not equal: 
        	            	expected: []string{"gcp-us-central1", "gcp-us-east1", "gcp-us-west1"}
        	            	actual  : []string{"gcp-us-central1", "gcp-us-east1"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,5 +1,4 @@
        	            	-([]string) (len=3) {
        	            	+([]string) (len=2) {
        	            	  (string) (len=15) "gcp-us-central1",
        	            	- (string) (len=12) "gcp-us-east1",
        	            	- (string) (len=12) "gcp-us-west1"
        	            	+ (string) (len=12) "gcp-us-east1"
        	            	 }
        	Test:       	TestSQLStatsRegions/secondary_tenant/distsql=false
        --- FAIL: TestSQLStatsRegions/secondary_tenant/distsql=false (6.31s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsRegions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45585

pkg/ccl/testccl/sqlstatsccl/sqlstatsccl_test.TestSQLStatsRegions [failed](https://tanzanite.cluster.engflow.com/invocations/default/b67912c1-6ffa-4d88-886c-df99b7ff8c7b?testReportRun=9&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3Fsc3RhdHNjY2w6c3Fsc3RhdHNjY2xfdGVzdA==) on master @ [49cff91f3501494deaf038671bc643c194a0e3ca](https://github.com/cockroachdb/cockroach/commits/49cff91f3501494deaf038671bc643c194a0e3ca):


```
=== RUN   TestSQLStatsRegions
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestSQLStatsRegions3183814399
    test_log_scope.go:76: use -show-logs to present logs inline
    sql_stats_test.go:211: -- test log scope end --
test logs left over in: outputs.zip/logTestSQLStatsRegions3183814399
--- FAIL: TestSQLStatsRegions (110.53s)
=== RUN   TestSQLStatsRegions/secondary_tenant
    --- FAIL: TestSQLStatsRegions/secondary_tenant (11.19s)
=== RUN   TestSQLStatsRegions/secondary_tenant/distsql=false
    sql_stats_test.go:205: 
        	Error Trace:	pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:205
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:148
        	            				pkg/testutils/subtest.go:23
        	Error:      	Not equal: 
        	            	expected: []string{"gcp-us-central1", "gcp-us-east1", "gcp-us-west1"}
        	            	actual  : []string{"gcp-us-central1", "gcp-us-west1"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,3 @@
        	            	-([]string) (len=3) {
        	            	+([]string) (len=2) {
        	            	  (string) (len=15) "gcp-us-central1",
        	            	- (string) (len=12) "gcp-us-east1",
        	            	  (string) (len=12) "gcp-us-west1"
        	Test:       	TestSQLStatsRegions/secondary_tenant/distsql=false
        --- FAIL: TestSQLStatsRegions/secondary_tenant/distsql=false (7.64s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsRegions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/sqlstatsccl/sqlstatsccl_test.TestSQLStatsRegions [failed](https://tanzanite.cluster.engflow.com/invocations/default/5e068541-ddd1-4263-92d5-4389a1f15ad8?testReportRun=10&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3Fsc3RhdHNjY2w6c3Fsc3RhdHNjY2xfdGVzdA==) on master @ [fe048f6e72edca59b4367463db1d0d00783f08ab](https://github.com/cockroachdb/cockroach/commits/fe048f6e72edca59b4367463db1d0d00783f08ab):


```
=== RUN   TestSQLStatsRegions
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestSQLStatsRegions2795081845
    test_log_scope.go:76: use -show-logs to present logs inline
    sql_stats_test.go:211: -- test log scope end --
test logs left over in: outputs.zip/logTestSQLStatsRegions2795081845
--- FAIL: TestSQLStatsRegions (105.51s)
=== RUN   TestSQLStatsRegions/secondary_tenant
    --- FAIL: TestSQLStatsRegions/secondary_tenant (65.93s)
=== RUN   TestSQLStatsRegions/secondary_tenant/distsql=false
    sql_stats_test.go:205: 
        	Error Trace:	pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:205
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:148
        	            				pkg/testutils/subtest.go:23
        	Error:      	Not equal: 
        	            	expected: []string{"gcp-us-central1", "gcp-us-east1", "gcp-us-west1"}
        	            	actual  : []string{"gcp-us-central1", "gcp-us-west1"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,3 @@
        	            	-([]string) (len=3) {
        	            	+([]string) (len=2) {
        	            	  (string) (len=15) "gcp-us-central1",
        	            	- (string) (len=12) "gcp-us-east1",
        	            	  (string) (len=12) "gcp-us-west1"
        	Test:       	TestSQLStatsRegions/secondary_tenant/distsql=false
        --- FAIL: TestSQLStatsRegions/secondary_tenant/distsql=false (59.69s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsRegions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/sqlstatsccl/sqlstatsccl_test.TestSQLStatsRegions [failed](https://tanzanite.cluster.engflow.com/invocations/default/fa9e778e-f44b-4af6-a974-5bb4f8ce0134?testReportRun=4&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3Fsc3RhdHNjY2w6c3Fsc3RhdHNjY2xfdGVzdA==) on master @ [a653b4e4e6483cec7d65808ba4d55d8c63747a6e](https://github.com/cockroachdb/cockroach/commits/a653b4e4e6483cec7d65808ba4d55d8c63747a6e):


```
=== RUN   TestSQLStatsRegions
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestSQLStatsRegions3367689806
    test_log_scope.go:76: use -show-logs to present logs inline
    sql_stats_test.go:211: -- test log scope end --
test logs left over in: outputs.zip/logTestSQLStatsRegions3367689806
--- FAIL: TestSQLStatsRegions (150.10s)
=== RUN   TestSQLStatsRegions/secondary_tenant
    --- FAIL: TestSQLStatsRegions/secondary_tenant (52.43s)
=== RUN   TestSQLStatsRegions/secondary_tenant/distsql=false
    sql_stats_test.go:205: 
        	Error Trace:	pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:205
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:213
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go:148
        	            				pkg/testutils/subtest.go:23
        	Error:      	Not equal: 
        	            	expected: []string{"gcp-us-central1", "gcp-us-east1", "gcp-us-west1"}
        	            	actual  : []string{"gcp-us-central1", "gcp-us-west1"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,3 @@
        	            	-([]string) (len=3) {
        	            	+([]string) (len=2) {
        	            	  (string) (len=15) "gcp-us-central1",
        	            	- (string) (len=12) "gcp-us-east1",
        	            	  (string) (len=12) "gcp-us-west1"
        	Test:       	TestSQLStatsRegions/secondary_tenant/distsql=false
        --- FAIL: TestSQLStatsRegions/secondary_tenant/distsql=false (2.86s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsRegions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go
+++ b/pkg/ccl/testccl/sqlstatsccl/sql_stats_test.go
@@ -32,7 +32,6 @@ func TestSQLStatsRegions(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 137616)
 	skip.UnderDuress(t, "test is too heavy for special configs")
 
 	// We build a small multiregion cluster, with the proper settings for
@@ -145,68 +144,53 @@ func TestSQLStatsRegions(t *testing.T) {
 			db := tc.db
 			db.Exec(t, `SET CLUSTER SETTING sql.txn_stats.sample_rate = 1;`)
 
-			testutils.RunTrueAndFalse(t, "distsql", func(t *testing.T, distsql bool) {
-				testutils.SucceedsWithin(t, func() (err error) {
-					var expectedRegions []string
-					_, err = db.DB.ExecContext(ctx, fmt.Sprintf(`USE %s`, tc.dbName))
-					if err != nil {
-						return err
+			testutils.SucceedsWithin(t, func() (err error) {
+				var expectedRegions []string
+				_, err = db.DB.ExecContext(ctx, fmt.Sprintf(`USE %s`, tc.dbName))
+				if err != nil {
+					return err
+				}
+
+				// Use EXPLAIN ANALYSE (DISTSQL) to get the accurate list of nodes.
+				explainInfo, err := db.DB.QueryContext(ctx, `EXPLAIN ANALYSE (DISTSQL) SELECT * FROM test`)
+				if err != nil {
+					return err
+				}
+				for explainInfo.Next() {
+					var explainStr string
+					if err := explainInfo.Scan(&explainStr); err != nil {
+						t.Fatal(err)
 					}
 
-					if distsql {
-						_, err = db.DB.ExecContext(ctx, "SET distsql = on;")
-						if err != nil {
-							return err
-						}
-					} else {
-						_, err = db.DB.ExecContext(ctx, "SET distsql = off;")
-						if err != nil {
-							return err
-						}
-
-					}
-
-					// Use EXPLAIN ANALYSE (DISTSQL) to get the accurate list of nodes.
-					explainInfo, err := db.DB.QueryContext(ctx, `EXPLAIN ANALYSE (DISTSQL) SELECT * FROM test`)
-					if err != nil {
-						return err
-					}
-					for explainInfo.Next() {
-						var explainStr string
-						if err := explainInfo.Scan(&explainStr); err != nil {
-							t.Fatal(err)
-						}
-
+					explainStr = strings.ReplaceAll(explainStr, " ", "")
+					// Example str "  regions: cp-us-central1,gcp-us-east1,gcp-us-west1"
+					if strings.HasPrefix(explainStr, "regions:") {
+						explainStr = strings.ReplaceAll(explainStr, "regions:", "")
 						explainStr = strings.ReplaceAll(explainStr, " ", "")
-						// Example str "  regions: cp-us-central1,gcp-us-east1,gcp-us-west1"
-						if strings.HasPrefix(explainStr, "regions:") {
-							explainStr = strings.ReplaceAll(explainStr, "regions:", "")
-							explainStr = strings.ReplaceAll(explainStr, " ", "")
-							expectedRegions = strings.Split(explainStr, ",")
-							if len(expectedRegions) < len(regionNames) {
-								return fmt.Errorf("rows are not replicated to all regions."+
-									" Expected replication to following regions %s but got %s\n", regionNames, expectedRegions)
-							}
+						expectedRegions = strings.Split(explainStr, ",")
+						if len(expectedRegions) < len(regionNames) {
+							return fmt.Errorf("rows are not replicated to all regions."+
+								" Expected replication to following regions %s but got %s\n", regionNames, expectedRegions)
 						}
 					}
+				}
 
-					// Select from the table and see what statement statistics were written.
-					db.Exec(t, "SET application_name = $1", t.Name())
-					db.Exec(t, "SELECT * FROM test")
-					row := db.QueryRow(t, `
+				// Select from the table and see what statement statistics were written.
+				db.Exec(t, "SET application_name = $1", t.Name())
+				db.Exec(t, "SELECT * FROM test")
+				row := db.QueryRow(t, `
 				SELECT statistics->>'statistics'
 				  FROM crdb_internal.statement_statistics
 				 WHERE app_name = $1`, t.Name())
 
-					var actualJSON string
-					row.Scan(&actualJSON)
-					var actual appstatspb.StatementStatistics
-					err = json.Unmarshal([]byte(actualJSON), &actual)
-					require.NoError(t, err)
-					require.Equal(t, expectedRegions, actual.Regions)
-					return nil
-				}, 3*time.Minute)
-			})
+				var actualJSON string
+				row.Scan(&actualJSON)
+				var actual appstatspb.StatementStatistics
+				err = json.Unmarshal([]byte(actualJSON), &actual)
+				require.NoError(t, err)
+				require.Equal(t, expectedRegions, actual.Regions)
+				return nil
+			}, 3*time.Minute)
 		})
 	}
 }

